# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/Nav33dCodes/flyrank-ml-internship-naveed/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This notebook frames a provisional capstone lane using careful, decision-focused language and real numbers from the starter dataset.

## 1. My lane and why

**Provisional lane:** Content Refresh Prioritization

I am choosing the Content Refresh Prioritization lane. I want to study which existing pages should be reviewed and refreshed first based on their observed search performance. This lane is useful because a content team may have thousands of pages but limited time, so it needs a practical way to decide which pages deserve attention first.

**Research question:** Which pages should a content team review first for a possible content refresh based on declining search performance and available page-level signals?

**Unit of analysis:** One webpage.

**Expected output:** A ranked list of pages, with pages showing stronger evidence of declining performance placed nearer the top.

## 2. The question: decision, action, and cost of a wrong call

**Decision:** Decide which pages should receive priority for manual review and a possible content refresh.

**Who acts on it:** A content or SEO team.

**Action:** The team reviews the highest-ranked pages and decides whether to update, improve, consolidate, or leave each page unchanged.

**Cost of a wrong recommendation:** If a healthy page is incorrectly ranked as high priority, the team may waste time editing content that did not need attention. If a genuinely declining page is missed, the team may delay an important review and potentially lose more search traffic.

**Why data or ML can help:** A team cannot manually compare thousands of pages consistently. Data analysis or ML can combine several observed signals and prioritize pages for human review. The goal is to improve a decision, not merely to train a model, and the result should support rather than replace human judgment.

## 3. Quick look at the data (2–3 real numbers)

The code below loads the anonymized starter dataset and calculates supporting numbers for this lane.

In [1]:
import pandas as pd

url = "https://raw.githubusercontent.com/Nav33dCodes/flyrank-ml-internship-naveed/main/data/raw/content_refresh_anonymized.csv"
df = pd.read_csv(url)

total_pages = len(df)
declining_pages = (df["trend_direction"] == "down").sum()
declining_percentage = (df["trend_direction"] == "down").mean() * 100
median_impressions = df["impressions_90d"].median()

print("Total pages:", total_pages)
print("Pages with a declining trend:", declining_pages)
print("Percentage with a declining trend:", round(declining_percentage, 2), "%")
print("Median impressions in the last 90 days:", round(median_impressions, 2))


Total pages: 30000
Pages with a declining trend: 16262
Percentage with a declining trend: 54.21 %
Median impressions in the last 90 days: 731.0


### What these numbers suggest

The starter dataset contains **30,000 pages**. Of these, **16,262 pages**, or **54.21%**, are labelled with a declining trend. The median page received **731 impressions** during the measured 90-day period.

These observed numbers suggest that content refresh prioritization is worth investigating. There are many pages to assess, and more than half show a declining trend, so a ranked decision-support approach may help a team focus its limited review time. These numbers do not prove that refreshing a page will improve its performance.

## 4. Careful words: what I can and cannot claim

I can say that the dataset contains **observed and measured signals** that may help prioritize pages for review. I can describe relationships as **directional** and present the final ranking as **decision support** for a content or SEO team.

I cannot claim that this analysis proves how Google's ranking algorithm works. I also cannot claim that refreshing a recommended page will definitely improve its ranking, clicks, impressions, or traffic. The analysis will not establish causation, and its recommendations should be checked by a human before action is taken.

## 5. Self-check

- [x] I selected one provisional lane.
- [x] I named the unit of analysis, decision, output, action, and cost of a wrong call.
- [x] I showed real numbers from the starter dataset.
- [x] I explained why this is a decision problem rather than only “train a model.”
- [x] I used careful language such as observed, measured, directional, and decision-support.
- [x] I included no client names, URLs, or private queries.
- [x] The notebook has no failing code cells.